engine/src/test/java/com/sandboni/core/engine/finder/scanner/DirectoryScannerTest.java

```diff
@@ -93,4 +93,27 @@ public void skipDependencyLocations() {
             return result;
         }
     }
+
+    @Test
+    public void dependencyScanWhenEnablePreviewFalse() {
+        String[] srcLocation = new String[]{"srcLocation1", "srcLocation2"};
+        String[] testLocation = new String[]{"testLocation1", "testLocation2"};
+        String[] dependencies = new String[]{"dependency1", "dependency2"};
+        Set<String> allLocations = new HashSet<>();
+        allLocations.addAll(Stream.of(srcLocation).map(location -> new File(location).getAbsolutePath()).collect(Collectors.toSet()));
+        allLocations.addAll(Stream.of(testLocation).map(location -> new File(location).getAbsolutePath()).collect(Collectors.toSet()));
+
+        Context context = new Context("appId", srcLocation, testLocation, dependencies, "", new ChangeScopeImpl(), null, null, false);
+
+        Map<String, Set<File>> scanResult = directoryScanner.scan((location, files) -> {
+            // not called because is not running parallel executor
+        }, context, true);
+
+
+        assertEquals(4, scanResult.size());
+        allLocations.forEach(location -> {
+            assertTrue(scanResult.containsKey(location));
+            assertFalse(scanResult.get(location).isEmpty());
+        });
+    }
 }
```

Comment:
This is awesome. can we add another test with enable.preview=true to show the difference in behavior?

Reply:
Thanks for the comment. I believe that is already accounted for test beginning on line 32